### PR TITLE
Add Rspec Unit Tests

### PR DESCRIPTION
**Pull request by GeekyHacks**
In this PR we did the following:
- Installed the Rspec testing gem
- Added all inquired test cases Book Person, Student, Teacher, Classroom, and Rental
- Put them all on spec folder
- Made sure that all test cases passed
- Fix linter errors

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './person'
+
+describe Person.new 23, 'Abdu' do
+  it { is_expected.to have_attributes(age: 23) }
+  it { is_expected.to have_attributes(name: 'Abdu') }
+  it { is_expected.to have_attributes(parent_permission: true) }
+end
```
Comment (VICTOR-CHUKWU) [inline spec/person_spec.rb]: - [ ] Good work writing your unit tests, but you forgot to test methods in the class, methods like `can_use_services`, `correct_name`, `add_rental` also test `default value for the optional name parameter if it is not provided`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @GeekyHacks  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Classes tested correectly
## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      expect(@rental).to be_an_instance_of(Rental)
+    end
+
+    it 'adds rental to person rentals list when add_rental method is called' do
+      book = double('Book', rentals: [])
+      @rental = Rental.new('2020-10-10', book, @person)
+
+      expect(book.rentals).to include(@rental)
+    end
+  end
```
Comment (OLIPLICHE) [inline spec/person_spec.rb:67]: - [ ] Hi, as stated by the previous reviewer, you forgot to test the method like testing the `default value for the optional name parameter if it is not provided`. Can you add this test to complete the test requirement?

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- add unit tests 👍 
- Classes tested correctly 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**